dd-sdk-android/src/test/kotlin/com/datadog/android/utils/assertj/DeserializedTelemetryDebugEventAssert.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.utils.assertj
+
+import com.datadog.android.telemetry.model.TelemetryDebugEvent
+import org.assertj.core.api.AbstractAssert
+
+internal class DeserializedTelemetryDebugEventAssert(actual: TelemetryDebugEvent) :
```

Comment:
Why do you need this one, when there's already a `TelemetryDebugEventAssert`?

Reply:
To be consistent with other `DeserializedXXXAssert` classes which exist for other RUM events, even though these RUM events have `XXXAssert` classes as well. Do they (`DeserializedXXXAssert`) exist only to skip `context`/`usr`/`additionalProperties` fields from comparison?

Reply:
I don't know why those exist… I probably missed them in previous code reviews. IMHO we should only have one class, and if need be create distinct `equalsWithX` or `equalsWithoutY` methods to make it understandable.

Reply:
I've extracted the assertion code in https://github.com/DataDog/dd-sdk-android/pull/869/commits/22097e1fddfc518593360741c2663bc497bb477c. Using recursive comparison has one advantage over `TelemetryXXXAssert`: no need to do anything if new field is added (but readability in case of the failure is worse). And in this test actual and expected share the same type, so recursive comparison also makes sense.